query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -242,32 +242,29 @@ class MedianAbsoluteDeviationOverTimeFunction(funcParams: Seq[Any]) extends Rang
                     ): Unit = {
     val q = 0.5
     val values: Buffer[Double] = Buffer.ofSize(window.size)
-    var i = 0
-    while (i < window.size) {
+    for (i <- 0 until window.size) {
       val curValue = window.apply(i).getDouble(1)
       if (!curValue.isNaN) {
         values.append(curValue)
       }
-      i = i + 1
     }
-    val counter = values.length
+    val size = values.length
     values.sort(spire.algebra.Order.fromOrdering[Double])
-    val (weight, upperIndex, lowerIndex) = QuantileOverTimeFunction.calculateRank(q, counter)
+    val (weight, upperIndex, lowerIndex) = QuantileOverTimeFunction.calculateRank(q, size)
     var median : Double = Double.NaN
-    if (counter > 0) {
+    if (size > 0) {
       median = values(lowerIndex)*(1-weight) + values(upperIndex)*weight
     }
 
     var medianAbsoluteDeviationResult : Double = Double.NaN
     val diffFromMedians: Buffer[Double] = Buffer.ofSize(window.size)
 
-    while (i < window.size) {
+    for (i <-0 until window.size) {
```

Comment:
Add a space here between `<-` and `0` (Not 100% sure, but that might give a syntax error).